### ISSUE TITLE
Bug: Query is not working

### ISSUE THREAD

**Issue by Hansereee** [SolrNet/SolrNet#320]
I tried to query my solr instance. But I alway get an exception in solrnet that says: "Data at the root level is invalid"

Here is my code:

` Startup.Init(Of TestDok)("http://localhost:8983/solr/test")
Dim Solr As ISolrOperations(Of TestDok) = ServiceLocator.Current.GetInstance(Of ISolrOperations(Of TestDok))()
Dim p1 = Solr.Query(SolrQuery.All)
`
The name of the core is "test"

**Comment by xavier-morera** [comment]
@Hansereee can you post the full request? Maybe you can use Fiddler so I can take a look

**Comment by BkrBkr** [comment]
I had the same problem and investigated it:

If I use "Solr.Query" the solr server respondes with a JSON format by default. (I have the default solr config). SolrNet expects an XML format and will throw an exception.

If I add the parameter "wt" with the value "xml" everything works fine.

Here is an example:
Dim p as new Dictionary(Of String,String)
p.add("wt","xml");
Solr.Query(new SolrQuery(...),new QueryOptions with{.ExtractParams=p})

You may should add the "wt" param by default.

**Comment by xavier-morera** [comment]
@BkrBkr Good point. I need to check why this is not already done this way.

**Comment by amitmiran** [comment]
here is theh C# solution
```
var p= new Dictionary<string,string>();
            p.Add("wt", "xml");
            var queryOptions = new QueryOptions();
            queryOptions.ExtraParams = p;


            var flaggedEventsCore01S = solr.Query("*:*", queryOptions);
```

**Comment by gjunge** [comment]
@xavier-morera @BkrBkr 
I believe that the default Solr response is in `xml`, I guess that is why it was never passed along. 

Now the question is where to put this fix: I think the place would be (Post)`SolrConnection`. 

This will tie-in `SolrConnection` with XML, which is not so nice (theoretically , the connection should not know about the data type being passed). But on the other hand, there are already places within the (Post)`SolrConnection` where there is a connection with Xml , namely in the Post methods, and also the "version" query string parameter is passed in which belongs to the Xml Response Writer. (https://lucene.apache.org/solr/guide/6_6/response-writers.html#ResponseWriters-TheversionParameter)
When we want to support other formats, we will fix the SolrConnection (or create a different `ISolrConnection`)

**Comment by BkrBkr** [comment]
@gjunge
I have the default solr installation with the default solr config files. I get an json formated response by default.

**Comment by gjunge** [comment]
@bkrbkr probably you’re running SOLR 7. Since SOLR 7 they’ve changed the default response type from XML to JSON , this effectively breaking SolrNet. 

@xavier-morera we should release the next version soon because SOLR 7 cheers contains a breaking change which was fixed by#322

**Comment by xavier-morera** [comment]
@gjunge I can do so immediately but there is an issue with Castle in develop branch. Would it be possible for you to take a look and I will release immediately. Thanks

**Comment by gjunge** [comment]
@xavier-morera I’ll have a look in a little bit. Could you share the url of your AppVeyor build?

**Comment by gjunge** [comment]
@xavier-morera actually it all compiles, it's just some weird test  that fails (a unit test on some performance test classes [which are  skipped for now], so actually a test of a test), and only in AppVeyor, not on my home PC (and not on my work PC). 
I'm inclined on just commenting out these tests , but I'll dig a little further before doing so.

**Comment by gjunge** [comment]
@xavier-morera  Found the problem. AppVeyor is running tests on all .dll it's find. Unfortunately the SolrNet.test.dll is both a library with unit tests, and a kind of a common test library. 

So AppVeyor finds this dll (without its dependencies) in the AutoFac test directory:
![image](https://user-images.githubusercontent.com/194466/32344538-dfc1b770-c00f-11e7-9688-f4077b5316f3.png)
and tries to run it.

So bottom line, the tests fail because of AppVeyor, and it's save to deploy. 
Let me just see how much work it will be to "debundle" this. I'll let you know soon.

**Comment by xavier-morera** [comment]
@gjunge great. Let me know if you can debundle as I am pushing to Nuget from AppVeyor and it won't create packages until build passes. I know I could build them manually but if this problem can be solved once and for all it would be better.

**Comment by gjunge** [comment]
@xavier-morera It's coming up.

**Comment by xavier-morera** [comment]
@gjunge it's building

**Comment by xavier-morera** [comment]
Fix deployed to Nuget

![image](https://user-images.githubusercontent.com/7174278/32350250-75e9de70-bfdf-11e7-9f73-412e71e5dbf5.png)

**Comment by BkrBkr** [comment]
@gjunge Yes I use SOLR 7
